### PR TITLE
Feature/members

### PR DESCRIPTION
**Pull request by Larkgronick**
Request date:
12/02/21

Total:
• Created members, members progress and members tasks pages
• Add drawer to navigate
• Implemented Router for these pages
• Realized own design

### CODE REVIEW

File: src/components/Drawer/Drawer.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  { name: 'Members', path: '/members', img: membersIcon },
+  { name: 'Tasks', path: '/tasks', img: tasksIcon },
+  { name: 'Progress', path: '/progress', img: progressIcon },
+];
+
+export function Drawer({ showDrawer, toggle }) {
+  return (
+    <aside className={showDrawer ? 'side-drawer open' : 'side-drawer'}>
+      <img className='dev-logo' src={devLogo} alt='dev-incubator-logo' />
+      {menuItems.map((item) => (
```
Comment (maximsan) [inline src/components/Drawer/Drawer.jsx:19]: spread item :)

File: src/config.js [hunk 2]
```diff
@@ -1,2 +1,2 @@
 export const appTitle = process.env.REACT_APP_DIMS_TITLE || 'Dims App';
-export const appLogo = process.env.REACT_APP_CUSTOM_LOGO;
+// export const appLogo = process.env.REACT_APP_CUSTOM_LOGO;
```
Comment (maximsan) [inline src/config.js:2]: remove

File: src/pages/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          <button className='header-button' type='button'>
+            Login
+          </button>
+        </div>
+      </header>
+      <main className='login-form'>
+        <h2>
+          Welcome<span>back!</span>
+        </h2>
+        <form className='input-fields' action=''>
```
Comment (maximsan) [inline src/pages/Login.jsx:26]: remove action

File: src/pages/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    <div className='login'>
+      <header className='header-login'>
+        <img className='dev-logo-login' src={devLogo} alt='dev-incubator-logo' />
+        <div>
+          <button className='header-button' type='button'>
+            Register
+          </button>
+          <button className='header-button' type='button'>
+            Login
+          </button>
```
Comment (maximsan) [inline src/pages/Login.jsx:19]: use Button component

File: src/pages/Main.jsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import { Route } from 'react-router-dom';
+import { Login } from './Login';
+import { Members } from './Members';
+import { Tasks } from './Tasks';
+import { Progress } from './Progress';
+
+export function Main({ showDrawer, toggle }) {
+  return (
+    <main className={showDrawer ? 'drawer-open' : ''}>
+      <Route path='/' component={() => <Login />} />
```
Comment (maximsan) [inline src/pages/Main.jsx:12]: ```<Route path='/' component={Login} />```

File: src/pages/Members.jsx [hunk 6]
```diff
@@ -1,4 +1,81 @@
...
+    direction: '.NET',
+    education: 'BSU',
+    education_img: educationIcon,
+    start: '30.01.2021',
+    start_img: startIcon,
+    age: '22',
+    e_mail: 'johndoe2@design.com',
+    buttons: 'buttons',
+  },
+];
```
Comment (maximsan) [inline src/pages/Members.jsx:33]: move constants to separate file

**Review by maximsan — CHANGES_REQUESTED** [review]
Hello, I left some comments, review them. Don't forget to rewrite your styles. Thanks!

File: src/pages/Progress.jsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+export function Progress({ showDrawer, toggle }) {
+  return (
+    <article>
+      <header className='header'>
+        <Hamburger showDrawer={showDrawer} toggle={toggle} />
+      </header>
+      <p className='page-name'>
+        {`${memberName}'s ${pageName}`}
+        <span>{`(${ProgressBody.length})`}</span>
```
Comment (YauhenKoipish) [inline src/pages/Progress.jsx:45]:
```suggestion
     <span>{ProgressBody.length}</span>
```
it looks better

File: src/pages/Tasks.jsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  const { showDrawer, toggle } = props;
+  return (
+    <article>
+      <header className='header'>
+        <Hamburger showDrawer={showDrawer} toggle={toggle} />
+        <Button name={pageName} />
+      </header>
+      <p className='page-name'>
+        {pageName}
+        <span>{`(${TasksBody.length})`}</span>
```
Comment (YauhenKoipish) [inline src/pages/Tasks.jsx:42]:
```suggestion
        <span>{TasksBody.length}</span>
```
it looks better

File: src/pages/Tasks.jsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+      </p>
+      <table className='table'>
+        <thead className='table-head'>
+          {menuItems.map((item) => (
```
Comment (YauhenKoipish) [inline src/pages/Tasks.jsx:46]: spread item.

File: src/pages/Tasks.jsx [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+Tasks.propTypes = {
+  showDrawer: PropTypes.func,
+  toggle: PropTypes.bool,
+};
+
+Tasks.defaultProps = {
+  showDrawer: PropTypes.func,
+  toggle: PropTypes.bool,
```
Comment (YauhenKoipish) [inline src/pages/Tasks.jsx:78]: You need to use default value, at default props

File: src/pages/Progress.jsx [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+          ))}
+        </tbody>
+      </table>
+    </article>
+  );
+}
+
+Progress.propTypes = {
+  showDrawer: PropTypes.func.isRequired,
+  toggle: PropTypes.bool.isRequired,
```
Comment (YauhenKoipish) [inline src/pages/Progress.jsx:73]: You need to use default value, at default props

File: src/App/App.jsx [hunk 12]
```diff
@@ -1,19 +1,29 @@
...
-import classes from './App.module.css';
+import { Component } from 'react';
+import { Drawer } from '../components/Drawer/Drawer';
+import { Main } from '../pages/Main';
 
-export const App = () => {
-  useEffect(() => {
-    document.title = appTitle;
-  }, []);
+export class App extends Component {
```
[the remark this answers is not in the document]
Comment (Larkgronick) [reply src/App/App.jsx]: Changed, thanks

File: src/components/Buttons/Button/Button.css [hunk 13]
```diff
@@ -0,0 +1,17 @@
+:root {
+  --text: #fff;
+  --dev-color: #008f02;
```
[the remark this answers is not in the document]
Comment (Larkgronick) [reply src/components/Buttons/Button/Button.css]: Changed to scss variables

File: src/components/Buttons/Hamburger/Hamburger.css [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+.bt-menu-trigger {
+  border: none;
+  outline: none;
+  background-color: var(--intervals);
+  font-size: 15px;
+  position: relative;
+  display: inline-block;
+  width: 2em;
```
[the remark this answers is not in the document]
Comment (Larkgronick) [reply src/components/Buttons/Hamburger/Hamburger.css]: Changed to rem

File: src/components/Buttons/Hamburger/Hamburger.css [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+  display: block;
+  width: 100%;
+  height: 0.2em;
+  margin-top: -0.1em;
+  background-color: var(--strokes);
+  -webkit-touch-callout: none;
+  -webkit-user-select: none;
+  -khtml-user-select: none;
+  -moz-user-select: none;
+  -ms-user-select: none;
```
[the remark this answers is not in the document]
Comment (Larkgronick) [reply src/components/Buttons/Hamburger/Hamburger.css]: Removed

File: src/components/Buttons/Hamburger/Hamburger.css [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+.bt-menu-trigger span {
+  position: absolute;
+  top: 50%;
+  left: 0;
+  display: block;
+  width: 100%;
+  height: 0.2em;
+  margin-top: -0.1em;
```
[the remark this answers is not in the document]
Comment (Larkgronick) [reply src/components/Buttons/Hamburger/Hamburger.css]: Changed to rem

File: src/components/Buttons/Hamburger/Hamburger.jsx [hunk 17]
```diff
@@ -0,0 +1,15 @@
+import PropTypes from 'prop-types';
+import './Hamburger.css';
+
+export function Hamburger({ showDrawer, toggle }) {
+  return (
+    <button onClick={toggle} className={showDrawer ? 'bt-menu-trigger bt-menu-open' : 'bt-menu-trigger'} type='button'>
```
[the remark this answers is not in the document]
Comment (Larkgronick) [reply src/components/Buttons/Hamburger/Hamburger.jsx]: Moved to Button component and replaced class

File: src/components/Drawer/Drawer.jsx (hunk 1 above)
Comment (Larkgronick) [reply src/components/Drawer/Drawer.jsx]: Changed, useful

File: src/config.js (hunk 2 above)
Comment (Larkgronick) [reply src/config.js]: Removed

File: src/pages/Login.jsx (hunk 3 above)
Comment (Larkgronick) [reply src/pages/Login.jsx]: Removed

File: src/pages/Login.jsx (hunk 4 above)
Comment (Larkgronick) [reply src/pages/Login.jsx]: Change both: register and login

File: src/pages/Main.jsx (hunk 5 above)
Comment (Larkgronick) [reply src/pages/Main.jsx]: Changed

File: src/pages/Members.jsx (hunk 6 above)
Comment (Larkgronick) [reply src/pages/Members.jsx]: Moved to constants.js. Made for every component.

File: src/pages/Progress.jsx [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+      <header className='header'>
+        <Hamburger showDrawer={showDrawer} toggle={toggle} />
+      </header>
+      <p className='page-name'>
+        {`${memberName}'s ${pageName}`}
+        <span>{`(${ProgressBody.length})`}</span>
+      </p>
+      <table className='table'>
+        <thead className='table-head'>
+          {menuItems.map((item) => (
```
[the remark this answers is not in the document]
Comment (Larkgronick) [reply src/pages/Progress.jsx]: Used spread for table rows

File: src/pages/Progress.jsx (hunk 7 above)
Comment (Larkgronick) [reply src/pages/Progress.jsx]: Here i need brackets - (3), your proposal makes only number - 3

File: src/pages/Tasks.jsx (hunk 8 above)
Comment (Larkgronick) [reply src/pages/Tasks.jsx]: Like in previous case i need brackets here

File: src/pages/Tasks.jsx (hunk 9 above)
Comment (Larkgronick) [reply src/pages/Tasks.jsx]: Changed

File: src/pages/Tasks.jsx (hunk 10 above)
Comment (Larkgronick) [reply src/pages/Tasks.jsx]: Changed to isRequered here

File: src/pages/Progress.jsx (hunk 11 above)
Comment (Larkgronick) [reply src/pages/Progress.jsx:42]: Here all is ok

File: src/pages/Progress.jsx (hunk 7 above)
Comment (YauhenKoipish) [reply src/pages/Progress.jsx]:
```suggestion
     <span>({ProgressBody.length})</span>
```
it looks more simple

File: src/pages/Tasks.jsx (hunk 8 above)
Comment (YauhenKoipish) [reply src/pages/Tasks.jsx]:
```suggestion
     <span>({ProgressBody.length})</span>
```
it looks more simple

File: src/pages/Progress.jsx (hunk 11 above)
Comment (YauhenKoipish) [reply src/pages/Progress.jsx:42]: default props is used for the default value, but you have no value

**Review by YauhenKoipish — APPROVED** [review]

File: src/pages/Tasks.jsx (hunk 8 above)
Comment (Larkgronick) [reply src/pages/Tasks.jsx]: Changed like in previous case

File: src/pages/Progress.jsx (hunk 7 above)
Comment (Larkgronick) [reply src/pages/Progress.jsx]: Yeah, thats much accurate, thanks. Changed in your way

**Review by YauhenKoipish — APPROVED** [review]

**Review by maximsan — APPROVED** [review]
Looks fine. I'm merging it.

**Pull request merged**